aperoll/star_field_items.py

```diff
@@ -187,34 +362,382 @@ def setPos(self, x, y):
+
+    def set_pos_for_attitude(self, attitude):
+        """
+        Set the position of all items in the field of view for the given attitude.
+
+        This method is called when the attitude of the scene changes. The position of all items is
+        recalculated based on the new attitude.
+
+        Parameters
+        ----------
+        attitude : Quaternion
+            The attitude of the scene.
+        """
+        if self.camera_outline is not None:
+            self.camera_outline.set_pos_for_attitude(attitude)
+        self._set_centroid_pos_for_attitude(attitude)
+
+    def _set_centroid_pos_for_attitude(self, attitude):
+        yag, zag = self._centroids["YAGS"], self._centroids["ZAGS"]
+        if attitude == self.attitude:
+            x, y = star_field_position(yag=yag, zag=zag)
+        else:
+            # `self.attitude` is the attitude represented by this field of view, but the scene's
+            # attitude is `attitude`. We first project back to get the ra/dec pointed to by the
+            # centroids using `self.attitude`, and calculate the position in the scene coordinate
+            # system using those ra/dec values
+            ra, dec = yagzag_to_radec(yag, zag, self.attitude)
+            x, y = star_field_position(attitude=attitude, ra=ra, dec=dec)
+
+        self.set_show_centroids(self._centroids_visible)
+
+        for i, centroid in enumerate(self.centroids):
+            centroid.setPos(x[i], y[i])
+
+    def boundingRect(self):
+        # this item draws nothing, it just holds children, but this is a required method.
+        return QtC.QRectF(0, 0, 1, 1)
+
+    def paint(self, _painter, _option, _widget):
+        # this item draws nothing, it just holds children, but this is a required method.
+        pass
+
+    def set_centroids(self, centroids):
+        """
+        Set the centroid values (usually from telemetry).
+
+        Parameters
+        ----------
+        centroids : astropy.table.Table
+            A table with the following columns: IMGNUM, AOACMAG, YAGS, ZAGS, IMGFID.
+        """
+        missing_cols = {"IMGNUM", "AOACMAG", "YAGS", "ZAGS", "IMGFID"} - set(
+            centroids.dtype.names
+        )
+        if missing_cols:
+            raise ValueError(f"Missing columns in centroids: {missing_cols}")
+
+        cols = list(centroids.dtype.names)
+        for col in cols:
+            self._centroids[col] = centroids[col]
+        self._set_centroid_pos_for_attitude(self.scene().attitude)
+
+    def set_show_centroids(self, show=True):
+        self._centroids_visible = show
+        # centroids are hidden if they fall outside the CCD
+        row, col = yagzag_to_pixels(
+            self._centroids["YAGS"], self._centroids["ZAGS"], allow_bad=True
+        )
+        off_ccd = (
+            (row < -511)
+            | (row > 511)
+            | (col < -511)
+            | (col > 511)
+            | (self._centroids["IMGFID"])
```

Comment:
Why are the fid light centroids excluded here?  They are a bit less interesting, but if we're just trying to display where on the CCD stuff is happening it might make sense to just include them.

Reply:
yes, I will add them back. I will use the same symbol but in red.